### ISSUE TITLE
How to build in debug?

### ISSUE THREAD

**Issue by sogartar** [openxla/xla#3381]
How should I go about building in debug? I am having trouble getting debug symbols into the binaries.
I tried
```
export CC=$(which gcc-10)
bazel build \
  --config=monolithic \
  --config=dbg \
  --spawn_strategy=local \
  --copt=-g \
  --linkopt=-g \
  --strip=never \
  //xla/tools:run_hlo_module
```

When I try loading the with `gdb` I get
```
$ gdb bazel-bin/xla/tools/run_hlo_module
GNU gdb (GDB) 12.1

....

(No debugging symbols found in bazel-bin/xla/tools/run_hlo_module)
```

My OS is Ubuntu 20.04.

**Comment by sogartar** [comment]
I tracked the problem that `-g0` is added to the compiler flags at the end after the flag coming from `--copt=-g`. This overrides it and the object file ends up with no debug info. https://gcc.gnu.org/onlinedocs/gcc/Debugging-Options.html

**Comment by sogartar** [comment]
The culprit where these two lines https://github.com/openxla/xla/blob/f487a876d84b94c66974670ad45da35180e8f485/.bazelrc#L276

This seems a remnant from when XLA was moved out of the TensorFlow repo.

**Comment by reedwm** [comment]
@sogartar, good catch on finding the culprit. I think all the the extra flags being passed with `--config=dbg` are unnecessary for the XLA repo, other than `-c dbg`. I tried building and running with `-c dbg` and I was able to debug and set breakpoints in gdb.

```
$ bazel build -c dbg //xla/tools:run_hlo_module
$ gdb --args bazel-bin/xla/tools/run_hlo_module --platform=cpu ~/dot.hlo
Copyright (C) 2022 Free Software Foundation, Inc.
...
(gdb) b run_hlo_module_main.cc:main
Breakpoint 1 at 0x88724e: file xla/tools/run_hlo_module_main.cc, line 73.
(gdb) r
...
Breakpoint 1, main (argc=3, argv=0x7fffffffe298) at xla/tools/run_hlo_module_main.cc:73
```

To get `--config=dbg` to work, we should remove these lines in XLA's .bazelrc file:

https://github.com/openxla/xla/blob/f487a876d84b94c66974670ad45da35180e8f485/.bazelrc#L270-L281

Unfortunately, I think XLA's .bazelrc is mirrored to TF's .bazelrc file, and TF needs these lines. @ddunl, is there a way to remove these lines only from XLA's .bazelrc file?
